src/components/TicketsTable.tsx

```diff
@@ -0,0 +1,80 @@
+import { DataTable } from 'primereact/datatable';
+import { Column } from 'primereact/column';
+import Link from 'next/link';
+import type { ServiceRequestSummary } from '@types';
+
+export interface TicketsTableProps {
+  items: ServiceRequestSummary[]
+}
+
+function OwnerAndPetTemplate({ client, pet, id }) {
+  return (
+    <div key={id}>
+      <Link href={`?ticket=${id}`}>
+        <div className="font-bold">{pet}</div>
+        <div className="capitalize">{client}</div>
+      </Link>
+
+    </div>
+  );
+}
+
+function CreatedAtTemplate({ created_at, id }) {
+  return (
+    <span key={id}>
+      <Link href={`?ticket=${id}`}>
+        {new Date(created_at).toLocaleDateString('en-US', {
+          day: '2-digit', month: '2-digit', year: 'numeric',
+        })}
+      </Link>
+    </span>
+  );
+}
+
+function descriptionView({ id, description }) {
+  return (
+    <Link href={`?ticket=${id}`} key={id}>
+      {description}
+    </Link>
+  );
+}
+function catergoryView({ id, category }) {
+  return (
+    <Link href={`?ticket=${id}`} key={id}>
+      {category}
+    </Link>
+  );
+}
+function TeamMemberView({ id, team_member }) {
+  return (
+    <Link className="text-red-500" href={`?ticket=${id}`} key={id}>
+      {team_member}
+    </Link>
+  );
+}
+
+function TicketsTable({ items }: TicketsTableProps, i) {
+  console.log('items', items);
+  return (
+    <DataTable
+      value={items}
+      paginatorTemplate="FirstPageLink PrevPageLink PageLinks NextPageLink LastPageLink CurrentPageReport RowsPerPageDropdown"
+      dataKey="id"
+      paginator
```

Comment:
This pagination might work while the dataset is small, but I think we will quickly run into issues as we have more tickets. This medium article talks about it: https://medium.com/@kannankannan18/client-side-pagination-vs-server-side-pagination-576a8f57257d

As they suggest, I think what we will need is a combination of both server and client pagination.

Reply:
You could also cheat and have the table titled "100 most recent requests". That way, there's not confusion when there are more requests than what was queried

Reply:
- Well these tables should not be rendering 100 records each time.  Maybe 20 at the most. 
- These tables will be handling closetickets, alltask, and  mytask. If you look at Kate's Design they are rendering 10-11 at most... Imagining if we multiply that by 10x for each? 
- Also I do not see the benefits of having server-side pagination. What are the benefits exactly? because we are only doing one query on page loading.

Reply:
I was thinking that the frontend would request 100 rows, but only 10 would be shown on the table at any time. I think this might be confusing because I'm using page in two different ways here. First, there is the "page" that's shown to the user on the frontend. That would be like 10 elements, keeping in line with Kate's design. Second, is the larger "page" included in the query, maybe 100 elements and not all of it is rendered at the same time.

Imagine if there were 30,000 records added to the db in a month. The user should be able to page through all of them. I think what should happen is the table would tell the user that there are 3,000 pages in the table. It would then make one request to get the first 100 rows. These rows would be broken into 10 frontend pages on the table. Once the user has clicked past all of those, the frontend would need to request the next 100 rows, which would continue to be broken down on the front end.

Does this make sense? I'm happy to set up a call to explain more.

Reply:
@DillonOLeary you are correct that if we want to scale we need server pagination. But this is just the react component, which has its own pagination concept. We don't need to bind the client page size to the server page. Please see my other comment in ClientService and let me know your thoughts.

Reply:
Prime's DataTable supports changeable row sizes, 
add to the table
`          
            rows={rows}
            rowsPerPageOptions={[10, 20, 50]}
`
add to the paginatorTemplate `RowsPerPageDropdown`

Add to the component:
`const [rows, setRows] = useState<number>(10);`

This will perform server side paginated queries 
```
async query(query: QueryModel): Promise<PageInfo<ServiceRequestType>> {
     const offset = query.page ? query.page * query.pageSize : 0;
     return supabaseClient.from('service_requests')
       .select('*', { count: 'exact' })
       .range(offset, offset + query.pageSize - 1)
       .order(query.sortField, { ascending: query.sortDirection === 'asc' })
       .then((resp) => ({
         rows: resp.data as ServiceRequestType[],
         totalRowCount: resp.count || 0,
       }));
   }
```

where:

```
type QueryModel = {
   page: number
   pageSize: number,
   sortField: string,
   sortDirection: string
 };
```
and 

```
type PageInfo<T> = {
   totalRowCount: number
   rows: T[]
 };
```

Reply:
> I was thinking that the frontend would request 100 rows, but only 10 would be shown on the table at any time. I think this might be confusing because I'm using page in two different ways here. First, there is the "page" that's shown to the user on the frontend. That would be like 10 elements, keeping in line with Kate's design. Second, is the larger "page" included in the query, maybe 100 elements and not all of it is rendered at the same time.
> 
> Imagine if there were 30,000 records added to the db in a month. The user should be able to page through all of them. I think what should happen is the table would tell the user that there are 3,000 pages in the table. It would then make one request to get the first 100 rows. These rows would be broken into 10 frontend pages on the table. Once the user has clicked past all of those, the front end would need to request the next 100 rows, which would continue to be broken down on the front end.
> 
> Does this make sense? I'm happy to set up a call to explain more.

Yes, you make sense. I was the one thinking about a totally different thing and was small in comparison. And it does do that, the limitation concerns you have was on tied with the initial limitation on the query for demo purposes.  Now we are on the same page with server-side pagination. Jeff has it above and the link I gave you also.

Reply:
> 

> 
> This will perform server side paginated queries 
> 


Is that cursor pagination along with offsets? Usually you have to choose one or the other. Really cool that Supabase supports both simultaneously!

Reply:
Supabase is stateless, so it can't hold on to an open cursor.  The offset is used to indicate which row to start extracting data.  Since there is no session (state), consecutive paging queries can "off" because of another user adding or removing records.

Reply:
Ah thanks for explaining. Didn't know Supabase was stateless. I misinterpreted the query. The .range() function still means a scan is done to find the staring row then, I was thinking it would start the scan at the offset. Anyways, good to know!